src/blockmap.jl

```diff
@@ -92,7 +92,7 @@ julia> L * ones(Int, 6)
 """
 Base.hcat
 
-function _hcat(As::Union{LinearMap,UniformScaling}...)
+function Base.hcat(As::Union{LinearMap,UniformScaling,AbstractVecOrMat}...)
```

Comment:
The AbstractVectors are here to avoid a method ambiguity with LinearAlgebra. This may be slightly controversial, because we don't ever interpret vectors as column matrices, but the use case here seems fairly specific, and why not interpret a vector in [h/v/hv]cat as a column of a bigger "matrix"?

Also, we haven't done this direct Base method overload originally because we were afraid that we may re-define Base behavior. But my understanding is that since our Union here is strictly bigger than what we have in Base/LinearAlgebra, our method is more generic, and hence does not get called when the arguments are only matrices and uniform scalings. I have added detailed tests for which method gets called when used without LinearMaps, and moreover, I am not getting any ambiguity errors, so I believe this should be fine.

And finally, this allows us to drop the requirement of having a LinearMap among the first 8 arguments. It seems like everything is just handled by dispatch.